src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalProductOpt.java

```diff
@@ -33,22 +34,20 @@ public LocalProductOpt(BaseDTO dto) {
     }
 
     public void getSupportedLocales(MessageCacheItem cacheItem) {
-        List<String> supportedLocales = new ArrayList<String>();
+        List<String> supportedLocales = new ArrayList<>();
         try {
-
             Path path = Paths.get(VIPCfg.getInstance().getOfflineResourcesBaseUrl());
-
-            URI uri = Thread.currentThread().getContextClassLoader().
-                    getResource(path.toString()).toURI();
-
-            if (uri.getScheme().equals("jar")) {
-                try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
-                    path = fileSystem.getPath(path.toString());
+            List<URI> uris = FileUtil.findUris(path);
+            for (URI uri : uris) {
+                if (uri.getScheme().equals("jar")) {
+                    try (FileSystem fileSystem = FileSystems.newFileSystem(uri, Collections.<String, Object>emptyMap())) {
+                        path = fileSystem.getPath(Paths.get(uri).toString());
+                        getSupportedLocales(path, supportedLocales);
+                    }
+                } else {
+                    path = Paths.get(uri);
                     getSupportedLocales(path, supportedLocales);
                 }
```

Comment:
Did the same mistake which causes a FileSystemNotFoundException. Because line 43 opens the file system, getSupportedLocales must be executed inside the try, before the file system closes.